autopush/endpoint.py

```diff
@@ -821,6 +830,38 @@ def put(self, uaid=""):
         d.addErrback(self._response_err)
         d.callback(uaid)
 
+    @cyclone.web.asynchronous
+    def delete(self, combo):
+        """HTTP DELETE
+
+        Invalidate a UAID (and all channels associated with it).
+
+        """
+        combo = combo.strip("/")
+        if "/" in combo:
+            (uaid, chid) = combo.split("/", 2)
+        else:
+            uaid = combo
+            chid = None
+        # what is request
+        if not self._validate_auth(uaid):
+            return self._write_response(
+                401, 109, message="Invalid Authentication")
+        message = self.ap_settings.message
+        if chid:
+            # mark channel as dead
+            self.ap_settings.metrics.increment("updates.client.unregister",
+                                               tags=self.base_tags())
+            d = deferToThread(message.delete_messages_for_channel, uaid, chid)
+            d.addCallback(message.unregister_channel, uaid, chid)
+            d.addErrback(self._response_err)
+            return d
+        # nuke uaid
+        d = deferToThread(message.delete_all_for_user, uaid)
+        d.addCallback(self.ap_settings.router.drop_user)
```

Comment:
Alternatively, you could inline a function here, that calls message.delete_all_for_user, and then calls drop_user, and punt that to deferToThread so that they'll run in the thread sequentially.